### ISSUE TITLE
[Question] Implementing into MVC

### ISSUE THREAD

**Issue by leijae** [Azure/communication-ui-library#543]
I have an existing MVC app; the only bundling it uses is the Bundler & Minifier extension in VS2019.... so the whole rush thing is new to me anyway. I thought it was like gulp... but apparently there also commands to run a iis express instance.. ok. 

I was able to successfully run the StaticHtmlComposite page in Node.js (which btw I'm not the biggest fan of since I work almost exclusively out of visual studio)

I installed the communication-react package from NPM.

I copied the packages that from the Node project for Calling only:
![image](https://user-images.githubusercontent.com/50371033/124832337-236b4400-df42-11eb-9148-6b2bd449603c.png)

I have the callComposite.js file from the StaticHtmlComposites sample linked to my view also and I get 
![image](https://user-images.githubusercontent.com/50371033/124833596-0d5e8300-df44-11eb-908c-24e9e8c4a4e6.png)

and the view looks like 

`
< div id="video-call" style="height: 750px"></ div>
<script src="/js/staticComposites/callComposite.js"></script>
@section Scripts
{
    <script type="text/javascript">
        var token = '';
        var user = '';

        function getToken() {
            var request = new XMLHttpRequest();
            request.open("GET", "api/calling");
            request.onreadystatechange = function() {
                if (this.readyState === 4 && this.status === 200) {
                    var responseObj = JSON.parse(this.response);
                    token = responseObj.token;
                    user = responseObj.user.communicationUserId;
                }
            };
            request.send();
        }
    </script>
    <script type="module">
    document.addEventListener("DOMContentLoaded", function (event) {
        getToken();
    });
    const callAdapter = await callComposite.loadCallComposite({
        containerId: 'video-call',
        groupId: '@Model.Content.GroupId', // Provide any GUID to join a group
        displayName: "VegetaIsBetterThanGoku",
        userId: userId,
        token: token
    });
    window.onbeforeunload = () => {
        callAdapter.dispose();
    };
</script>
}`

and it looks like the callComposite is breaking here
![image](https://user-images.githubusercontent.com/50371033/124833907-7b0aaf00-df44-11eb-84b8-5fd44395e7b2.png)

Probably I've got to build react at some point in my build process? I have no experience with that... I have nothing against React (unlike some in the community) I just have no experience with it. I'm not looking to add any skills to my portfolio haha, I'm passed that stage.

Man I'm frustrated. I get this is a client to client package, and I'm okay with JS SDKs but... it's not turn-key  enough for the likes of me. I just want to put the components on the view and theme them with my own css styles... lazy I know, but you guys made me that way :P with your awesome SDKs.

A little help please. I don't want to give up... but I'm close.

**Comment by JamesBurnside** [comment]
Unfortunately we do not have a hosting story regarding composite use outside of the npm package, that is something we are working on but won't be ready for some time.
Currently, to run the composites outside of react its a two steps process:
1. Build the static html composites from this repo - this generates two bundled js files:
   ```
   .\samples\StaticHtmlComposites\dist\chatComposite.js
   .\samples\StaticHtmlComposites\dist\callComposite.js
   ```
2. Copy and paste one or both of those files (depending on if you want the calling composite, chat composite or both) to your local project

Ideally you wouldn't need to do step 1 and the files could be obtained through npm or similar.

@PorterNan as you're making some changes in this area can you help out here? 

We also just updated our documentation (https://github.com/Azure/communication-ui-library/pull/539) on this very recently so feedback in this area on confusions is great

**Comment by leijae** [comment]
ok so I have that 
>.\samples\StaticHtmlComposites\dist\callComposite.js

Are you saying I should replace the other callComposite.js file with this one?

**Comment by PorterNan** [comment]
Hey, thanks for the question, so first thing first, if you look at the updated document, you might need to run `rush install` and `rush build` before you grab your js bundle, there is a strange error for now, that even you don't install these dependencies, js files will be generated but without those dependencies and no error was shown

Please check your js bundle file size, it should be ~1MB for chat

**Comment by leijae** [comment]
Holy crap I got it working.

Step 1) do the rush build thing.
Step 2) pull the new callComposite.js to project (i added it to js folder in wwwroot)
Step 3) I built a web api TokenController that looks almost exactly like the controller in the C# Calling Hero Sample out of the Calling.csproj/Controllers/UserTokenController

Step 4) Built a public MVC controller so that a link can be generated link to an active conversation may be shared and joined into  (i pass the groupId in through HTTPGET routing and username is automatically generated (or passed in) 

Interestingly THAT view is really basic

`@model ACSCommUILib.Controllers.PublicCallingViewModel
@{
    ViewData["Title"] = "Public Access Test";
}


<div id="video-call" style="height: 700px;"></div>
<script src="~/js/callComposite.js"></script>

@section Scripts
{
    <script type="module">
        setTimeout(function(){
            const callAdapter = callComposite.loadCallComposite({
                containerId: 'video-call',
                groupId: '@Model.GroupId', 
                displayName: "@Model.FirstName",
                userId: '@Model.UserCommunicationId',
                token: '@Model.Token'
            });
        }, 3000);

        window.onbeforeunload = () => {
            callAdapter.dispose();
        };

    </script>
}`

And the result
![image](https://user-images.githubusercontent.com/50371033/124857132-5da00a00-df71-11eb-8028-a9b1f083e1a9.png)



Step 4) Dance the funky chicken

I think the only left to do now is figure out if I can change the text "Start Call" to "Join Call" with some javascript...

but I did get it working in MVC thanks to your help. And yes ideally, the callComposite.js (bundled) would just be a straight NPM install instead of having to build it first, but I get it... this is still not a mature project and has many beta, previewish elements.

Thanks for the help!
